controllers/provisioning_controller.go

```diff
@@ -185,11 +185,15 @@ func (r *ProvisioningReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error
 		return ctrl.Result{}, err
 	}
 
-	if err := provisioning.ValidateBaremetalProvisioningConfig(baremetalConfig); err != nil {
+	if errs := provisioning.ValidateBaremetalProvisioningConfig(baremetalConfig); len(errs) > 0 {
 		// Provisioning configuration is not valid.
 		// Requeue request.
 		r.Log.Error(err, "invalid config in Provisioning CR")
```

Comment:
r.Log.Error(errs, "invalid config in Provisioning CR")

Reply:
oh yeah, good catch